agents/hwp_gripper/src/gripper.py

```diff
@@ -0,0 +1,287 @@
+                "Clearing Alarm group '%s' via a RESET" % (group))
+            return self.CTL.RESET()
+        elif group == "E":
+            self.log.err(
+                "RESET failed in Gripper.RESET() due to alarm group '%s' "
+                "detected. Power cycle of controller and motors required"
+                % (group))
+            return False
+        else:
+            self.log.err(
+                "RESET aborted in Gripper.RESET() due to unknown alarm group")
+            return False
+        if not self.ALARM():
+            self.log.log("Alarm successfully reset")
+            return True
+        else:
+            self.log.err(
+                "RESET aborted in Gripper.RESET() due to unknown error")
+            return False
+
+    def POSITION(self):
+        """ Print the gripper position """
+        for k in self.motors.keys():
+            self.log.out("Axis %s = %.02f mm" % (k, self.motors[k].pos))
+        return True
+
+    def SETPOS(self, axis_no, value):
+        """ Set a user-defined position for a specific motor """
+        self.log.out(
+            "Axis %d old position = %.02f" % (axis_no, self.motors[str(axis_no)].pos))
+        self.motors[str(axis_no)].pos = float(value)
+        self.motors[str(axis_no)].max_pos_err = 0.
+        self._write_pos()
+        self.log.out(
+            "Axis %d new position set manually = %.02f"
+            % (axis_no, value))
+        self.log.log(
+            "Axis %d maximum position error zerod" % (axis_no))
+        return True
+
+    def INP(self):
+        """ Return control INP """
+        outs = self.CTL.INP()
+        # for i in range(3):
+        #    self.log.out("INP%d = %d" % (i+1, outs[i]))
+        return outs
+
+    def STATUS(self):
+        """ Return control status """
+        return self.CTL.STATUS()
+
+    # ***** Helper Methods *****
+    def _read_pos(self):
+        """ Read motor positions from position file """
+        lines = self.posf.readlines()
+        if len(lines) == 0:
+            self.motors["1"].pos = 0.
+            self.motors["2"].pos = 0.
+            self.motors["3"].pos = 0.
+        else:
+            lastWrite = lines[-1]
+            date, time = lastWrite.split('[')[1].split(']')[0].split()
+            pos1, pos2, pos3 = lastWrite.split()[2:]
+            self.motors["1"].pos = float(pos1)
+            self.motors["2"].pos = float(pos2)
+            self.motors["3"].pos = float(pos3)
+        return True
+
+    def _write_pos(self, init=False):
+        """ Write motor positions to position file """
+        now = dt.datetime.now()
+        date = "%04d-%02d-%02d" % (now.year, now.month, now.day)
+        time = "%02d:%02d:%02d" % (now.hour, now.minute, now.second)
+        wrmsg = (
+            "[%s %s] %s %-20.2f %-20.2f %-20.2f\n"
+            % (date, time, ' '*3,
+               self.motors["1"].pos,
+               self.motors["2"].pos,
+               self.motors["3"].pos))
+        self.posf.truncate(0)
```

Comment:
This is new as well. So we don't really care to log all the positions, just the most recent one and you're using truncate to remove the file contents before writing the most recent position?